printf.h

```diff
@@ -43,12 +43,16 @@ extern "C" {
 /**
  * Lock the printf to avoid collision when multiple threads/cores try to output strings/characters.
  * This function can implement a mutex, semaphore or whatever lock.
+ *
+ * \note PRINTF_ENABLE_LOCK should be defined in order to activate lock function.
  */
 void __io_lock();

```

Comment:
Should these be wrapped in PRINTF_IO_LOCK preprocessor checks?

Reply:
Actually was using the format introduced by Marco to enable features.
Here `PRINTF_ENABLE_LOCK` is a macro to add in Makefile, and if it is defined, then `PRINTF_IO_LOCK` will be enabled and lock and unlock functions activated.